framework/include/restart/RestartableData.h

```diff
@@ -276,4 +320,14 @@ RestartableData<T>::loadInternal(std::istream & stream)
   loadHelper(stream, set(), _context);
 }
 
+template <typename T>
+inline void
+RestartableData<T>::storeJSONValue(nlohmann::json & json) const
+{
+  if constexpr (RestartableData<T>::has_store_json)
+    nlohmann::to_json(json, get());
+  else
```

Comment:
We could throw a compiler error here wit a static assert, what saith you?

Reply:
Nope. This method will still get instantiated for classes that don't have `to_json`, which is totally valid. We just can't call it. We could only do this if we want to require all restartable types to jave a `to_json` spec, which is no bueno.

Reply:
Okay, resolving then.